### ISSUE TITLE
Add custom unlock descriptions

### ISSUE THREAD

**Issue by Gonzom** [Js41637/Overwatch-Item-Tracker#219]
I know you get the unlocking text from the game itself, but I feel some context should be added for the ones the game does not currently have.

**Breast Cancer Research Foundation charity event**
"Ribbon" icon - "Unlocked by logging in during the Breast Cancer Research Foundation charity event"
"Ribbon" spray - "Unlocked by watching two cumulative hours of participating charity streams during the Breast Cancer Research Foundation charity event"
"Pink" icon and "Cure" spray - "Unlocked by watching four cumulative hours of participating charity streams during the Breast Cancer Research Foundation charity event"
"Together" spray and "Pink" spray - "Unlocked by watching six cumulative hours of participating charity streams during the Breast Cancer Research Foundation charity event"
"Pink" skin - "Unlocked by money purchase"

**Event Icons**
"Summer Games 2016" - "Unlocked by logging in during the 2016 Summer Games event"
"Summer Games 2017" - "Unlocked by logging in during the 2017 Summer Games event"
"Halloween Terror 2016" - "Unlocked by logging in during the 2016 Halloween Terror event"
"Halloween Terror 2017" - "Unlocked by logging in during the 2017 Halloween Terror event"
"Winter Wonderland 2016" - "Unlocked by logging in during the 2016 Winter Wonderland event"
"Winter Wonderland 2017" - "Unlocked by logging in during the 2017 Winter Wonderland event"
"Year of the Rooster 2017" - "Unlocked by logging in during the 2017 Lunar New Year event"
"Year of the Dog 2018" - "Unlocked by logging in during the 2018 Lunar New Year event"
"Uprising 2017" - Unlocked by logging in during the 2017 Overwatch Archives event"
"Archives 2018" - Unlocked by logging in during the 2018 Overwatch Archives event"

**Comment by Js41637** [comment]
I might add the breast cancer ones but I deliberately chose not to to the event ones as it's pretty obvious they're part of an event, icons are already small and adding text covers some of it up and feels redundant.

**Comment by Gonzom** [comment]
Fair enough. They do however act differently from the sprays, as you are able to buy the previous year spray one, but cannot earn the previous year icon one and they don't come from boxes, so I felt they should get an explanation, but I understand your point.

**Comment by Js41637** [comment]
Yeah it is a good point, i'll think about it.

**Comment by Js41637** [comment]
The issue with the Origins Edition skins is they already have descriptions. I don't have any other way to add a second description.

**Comment by Gonzom** [comment]
From a visual issue or coding issue?
If it's a visual issue then I understand, but if it's a code issue then maybe split the description and achievement text into 2 fields so you will be able to have 2 separate item text fields without overriding one.

**Comment by Js41637** [comment]
both

**Comment by Js41637** [comment]
I'm thinking maybe I should add any item descriptions to the image preview and then any type of unlock info to a tooltip when hovering over the event/achievement/special/etc icon.

**Comment by Gonzom** [comment]
Sounds like it could work

**Comment by Gonzom** [comment]
Added the Battle for Azeroth Collector's Edition emotes and voice lines to the list. The sprays and icons already have this.
